Graphics_Engine/Query/Color.cs

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Graphics;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Drawing;
+using System.Linq;
+
+namespace BH.Engine.Graphics
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****           Public Methods                  ****/
+        /***************************************************/
+
+        [Description("Gets a color from a gradient at the spacified position between 0 and 1")]
+        [Input("gradient", "The gradient to Query a Color from")]
+        [Input("val", "The number between 0 and 1 to use for interpolation of the markers colors")]
+        [Output("color", "The Color at the specified position")]
+        public static Color Color(this Gradient gradient, double val)
+        {
+            // keep val in the domain
+            val = val > 1 ? 1 : val < 0 ? 0 : val;
```

Comment:
not 100% sure if we want this or not, but could consider a warning message here if the value is being normalised.

Reply:
Sure? I think it's pretty standard to apply the edge value of the gradient to values outside of the range. 

Not doing it would mostly create some unpredictability if someone managed to get a marker in outside of 0 to 1, and given how common it is to set your from/to to other values outside your extreme I think warnings would mostly be annoying

Reply:
As stated, not sure :). And to be clear, I 100% agree with actually doing the normalisation, was just wondering if it would be worth letting the user know that it has been done or not. Basically, not sure if it is just annoying or useful do add a warning for it. Happy to not add it in now, can always add it in later if it feels unclear to anyone.